tidb-binlog/pump_client/pump.go

```diff
@@ -111,21 +117,36 @@ func (p *PumpStatus) createGrpcClient(security *tls.Config) error {
 	return nil
 }
 
-// closeGrpcClient closes the pump's grpc connection.
-func (p *PumpStatus) closeGrpcClient() {
+// ResetGrpcClient closes the pump's grpc connection.
+func (p *PumpStatus) ResetGrpcClient() {
+	p.Lock()
+	defer p.Unlock()
+
 	if p.grpcConn != nil {
 		p.grpcConn.Close()
 		p.Client = nil
 	}
 }
 
-func (p *PumpStatus) writeBinlog(req *pb.WriteBinlogReq, timeout time.Duration) (*pb.WriteBinlogResp, error) {
-	if p.Client == nil {
-		return nil, errors.Errorf("pump %s don't have avaliable pump client", p.NodeID)
+// WriteBinlog write binlog by grpc client.
+func (p *PumpStatus) WriteBinlog(req *pb.WriteBinlogReq, timeout time.Duration) (*pb.WriteBinlogResp, error) {
+	p.RLock()
```

Comment:
connection leak.

if p1 and p2 call `WriteBinlog` concurrently, they may create two client

Reply:
fix in 91ad124,  when create grpc client will close the old connection, so it dosen't matter